core/src/main/java/smile/regression/OLS.java

```diff
@@ -252,17 +285,26 @@ public OLS(double[][] x, double[] y, boolean SVD) {
                     coefficients[i][1] = se;
                     double t = w1[i] / se;
                     coefficients[i][2] = t;
-                    coefficients[i][3] = Beta.regularizedIncompleteBetaFunction(0.5 * df, 0.5, df / (df + t * t));
+                    coefficients[i][3] = Beta.regularizedIncompleteBetaFunction(0.5 * df, 0.5, df / (df + t * t)); 
                 } else {
                     coefficients[i][1] = Double.NaN;
                     coefficients[i][2] = 0.0;
                     coefficients[i][3] = 1.0;
                 }
+                if (!Math.isZero(s, .01)){
```

Comment:
Why 0.01? It seems very large to me. Do you have a reference on this choice?

Reply:
I used 0.01 because I was worried about 1/s^2 becoming very large, and potentially causing the elments in the gamma matrix becoming large and unstable. 

If you ignore the fact that I don't place 1/s^2 in sigmaInverse in cases where s is in (-0.01, 0.01), then it can be shown that the gamma matrix after the ith step is equal to (X^T * X)^-1, where X contains all initial input observations, and all input observations up to and including the ith step. 

So, the online learning solution after i learning steps is equivalent to finding coefficients with a traditional OLS solution. Maybe the best way to handle small values when initializing the gama matrix is to deal with them as you originally do in solve() methods in SVD and QR.